packages/react-components/src/shared/src/size.ts

```diff
@@ -1,15 +1,15 @@
 export type Size = "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl" | "3xl" | "4xl" | "5xl" | "inherit";
 
-export function normalizeSize(size?: Size) {
+export function normalizeSize<T extends Size>(size?: T) {
```

Comment:
doing this creates better adapters.

for this example
```
const adapter = createSizeAdapter({
"sm": "md",
"md": "lg"
})

const res = adapter("md");
```

before : 
res was of type Size

now: 
res is "md" | "lg"